### DISCUSSION TITLE
MD issue and questions

### DISCUSSION THREAD

**Discussion by lpizza** [grimme-lab/xtb#600] [Q&A]
Dear users, 

I encountered a serious issue when running molecular dynamics calculations of a silicon cluster, with Si surface atoms passivated by hydrogen atoms. In my calculations I need to keep physical vibrations of the Si-H bonds, therefore I deactived the SHAKE algorithm and I set the hmass parameter to 1.0. However the MD calculations stopped after several iterations (sometimes several hundreds) with 'Runtime exception occurred', the triggering error being that the 'Self consistent charge iterator did not converge'. It is a consequence of the MD algorithm becoming unstable. I know that with a low H mass, one cannot use large timestep, therefore I decreased the timestep, but the error still occurred even when using a timestep of 0.1 fs (which is pretty low). Therefore I turn to you whether you had suggestions to solve this issue? Just to mention, I also try to play with electronic temperature to improve scc convergence, but it seems that it is fixed to 300K in MD calculations (normally it shouldn't be important for my cluster, since the HOMO-LUMO gap is large). 

Other points I am wondering about:

- Is it possible to increase the printout frequency for MD? Information is output every 200 iterations, and I do not find a parameter to change the frequency. 
- Also is it possible to get information about the scc convergence at each MD step?
- There is a 'skip' parameter which can be defined, but apparently related to commandline options -'mdav' or '-mdopt', which are not documented?  

Thanks in advance for your help 

Laurent P. 
Pprime Institute - CNRS - Poitiers University, France

**Comment by haneug** [comment]
Dear Laurent,

I am not that familiar with MD simulations in xtb or MD simulations in general. 
However, this looks like you are running in some kind of hot spot here. Is the final structure where the SCC fails printed out?
Just because your cluster has a large HOMO-LUMO gap does not necessarily mean that you won't run into structures with a small HOMO-LUMO gap during your MD. 
Also, maybe one of the more robust GFN methods such as the forcefield GFN-FF or the non-SCC GFN0 are already sufficient for your needs?

best,

Hagen

**Reply by lpizza** [reply]
Dear Haneug, 

thanks for your reply. I am usually able to run several hundreds iterations before the issue occurred, so yes there are snapshots printed periodically before. One thing I can try, is print the structure at every MD step, which could help to see if for an unknown reason the atomic configuration went into a spurious state. I will try that. 
I agree that one could ran into a small HOMO-LUMO gap during the MD, but this should not occur in the system I consider, since the H-L gap of the stable structure is greater than 3 eV and the MD temperature is 300K. Also the fact that a Fermi smearing is applied should prevent any such issue. 
What is worrying (to me) is that I also performed DFT MD calculations for the same systems, without any issue. I am really interested in using the GFN-FF2 level of accuracy as a way to investigate larger system than with DFT-MD. Of course, I can use the force-field of the non-self-consistent level, but I am not sure the accuracy will be good enough for my purposes. 

Regarding the other points I mentioned in my message (the printout frequency in MD, the SCC information during MD, and the skip parameter), do you have some information ? 

Thanks and best regards

Laurent

**Reply by haneug** [reply]
> * Is it possible to increase the printout frequency for MD? Information is output every 200 iterations, and I do not find a parameter to change the frequency.

You can dump MD information. Take a look here: https://github.com/grimme-lab/xtb/blob/main/man/xcontrol.7.adoc#md

Regarding the other two points, I am afraid I can not help you. 

Alternatively you can use GFN2-xTB in other codes which lie more focus on MDs such as:
[ase](https://wiki.fysik.dtu.dk/ase/)
[dftbplus](https://dftbplus.org/)

best,

Hagen

**Reply by epunzal2** [reply]
Is GFNFF also in ase?
